hardware/pico_driver/scripts/pico_split_extended.py

```diff
@@ -57,25 +60,47 @@ def pico_xyz_from_extended(
     ext_topic = pico.get_ext_topic(cam)
     pts_topic = pico.get_pts_topic(cam)
     amp_topic = pico.get_amp_topic(cam)
-    with rosbag.Bag(out_bag_path, "w") as out_bag:
-        for ext_msg in pico.get_msgs(in_bag_path, ext_topic, fast):
-            distance, amplitude, intensity, noise = pico.split_extended(ext_msg)
 
-            xyz = pico.xyz_from_distance(distance, xyz_coeff)
-            pts_msg = pico.make_points_msg(ext_msg.header, xyz)
-            out_bag.write(pts_topic, pts_msg, pts_msg.header.stamp)
+    out_bag = rosbag.Bag(out_bag_path, "w")
 
-            amp_msg = pico.make_amp_msg(ext_msg.header, amplitude)
-            out_bag.write(amp_topic, amp_msg, amp_msg.header.stamp)
+    with rosbag.Bag(in_bag_path, "r") as bag:
+        # for ext_msg in pico.get_msgs(in_bag_path, ext_topic, fast):
+        for topic, msg, t in bag.read_messages():
```

Comment:
If you know you only want topics that constitute a small subset of the data, it is a major performance improvement to specify the topics in the read_messages() call because it can skip deserializing the other messages. That feature was used in the old version. (Here I guess it would be slightly more complicated because the topic list would only be used when --save-all-topics is not set.)